compiler/resolution/resolveFunction.cpp

```diff
@@ -1032,6 +1038,11 @@ static void insertUnrefForArrayOrTupleReturn(FnSymbol* fn) {
 
   Symbol* ret = fn->getReturnSymbol();
 
+  // surely returning neither tuple nor array, so return
+  if (ret->type != dtUnknown                   &&
+      ! ret->type->symbol->hasFlag(FLAG_ARRAY) &&
+      ! ret->type->symbol->hasFlag(FLAG_DOMAIN) ) return;
```

Comment:
Should this be checking for `FLAG_TUPLE` also, or `dtUnknown` implies tuple when we run this code?

Reply:
Indeed it should be checking FLAG_TUPLE. I added the check.